File: 1-basic/1600-problem-solving/150-topin/34-binary-tree-inorder-traversal-recursive-II/binary-tree-inorder-traversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *         this.val = (val === undefined ? 0 : val)
 *         this.left = (left === undefined ? null : left)
 *         this.right = (right === undefined ? null : right)
 * }
 */

 /**
  * @param {TreeNode} root
  * @return {number[]}
  */

  const inorderTraversal = root => {
      if (!root) {
          return [];
      }

      return [...inorderTraversal(root.left), root.val, ...inorderTraversal(root.right)];
  }

  root = {
      val : 1,
      left: {val: 2, left: {val: 4, left: null, right: null}, right: {val: 6, left: null, right: null}},
      right: {val: 3, left: {val: 5, left: null, rigth: null}, right: {val: 7, left: null, right: null}},
  }

  console.log(inorderTraversal(root))

File: 1-basic/1600-problem-solving/150-topin/34-binary-tree-inorder-traversal-recursive-II/binary-tree-inorder-traversal.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
    public:
    vector<int> vec;
    vector<int> inorderTraversal(TreeNode* root) {
        if(root == nullptr) return {};
        inorderTraversal(root->left);
        vec.push_back(root->val);
        inorderTraversal(root->right);
        return vec;
    }
}

int main() {
    return 0;
}